cpp/src/arrow/ipc/test_common.cc

```diff
@@ -79,6 +79,39 @@ Status MakeRandomInt32Array(int64_t length, bool include_nulls, MemoryPool* pool
   return Status::OK();
 }
 
+template <typename ArrayType>
+static Status MakeRandomArray(int64_t length, bool include_nulls, MemoryPool* pool,
+                              std::shared_ptr<Array>* out, uint32_t seed) {
+  random::RandomArrayGenerator rand(seed);
+  const double null_probability = include_nulls ? 0.5 : 0.0;
+
+  *out = rand.Numeric<ArrayType>(length, 0, 1000, null_probability);
+
+  return Status::OK();
+}
+
+template <>
+Status MakeRandomArray<Int8Type>(int64_t length, bool include_nulls, MemoryPool* pool,
```

Comment:
Make those static as well? (or put them in the anonymous namespace)

Reply:
To make this static cause the compilation error. I will put them into the anonymous namespace.

```
/home/ishizaki/Arrow/arrow/cpp/src/arrow/ipc/test_common.cc:94:1: error: explicit template specialization cannot have a storage class
 static Status MakeRandomArray<Int8Type>(int64_t length, bool include_nulls, MemoryPool* pool,
 ^~~~~~

```

Reply:
I will just remove `static` like other MakeRandom...Array.